src/components/auth/tenant/TenantRegistration.tsx

```diff
@@ -0,0 +1,129 @@
+"use client";
+
+import { useState } from 'react';
+import { useRouter } from "next/navigation";
+import { 
+  getAuth, 
+  createUserWithEmailAndPassword, 
+  sendEmailVerification 
+} from 'firebase/auth';
+import { useMutation } from '@apollo/client';
+import { ASSIGN_TENANT_ROLE, CREATE_USER_RECORD } from '@graphql/queries/userMutations.graphql';
+import PasswordStrengthIndicator from './PasswordStrengthIndicator';
+
+export const TenantRegistration: React.FC = () => {
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [error, setError] = useState<string | null>(null);
+  const router = useRouter();
+  
+  const [assignRole] = useMutation(ASSIGN_TENANT_ROLE);
+  const [createUser] = useMutation(CREATE_USER_RECORD);
```

Comment:
@rvalenciano hey 👋 would u mind doing an early a code review Im having this issue 
<img width="1049" alt="image" src="https://github.com/user-attachments/assets/8aed3d95-b979-4e40-bf9e-2b04e12ecf34">

I researched about it and seems to be something related to apollo setup but I'm not sure, do u have any idea what could it be?